.github/workflows/packer-build-and-publish.yml

```diff
@@ -233,11 +317,126 @@ jobs:
           fi
 
       - name: Publish release
+        id: publish
         uses: softprops/action-gh-release@c062e08bd532815e2082a85e87e3ef29c3e6d191 # v2.0.8
         with:
-          files: artifacts/**/*.json
+          files: |
+            artifacts/**/*.json
+            artifacts/**/*.md
           body: |
-            Release ${{ github.ref_name }} from workflow [#${{ github.run_id}}](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})
-          draft: ${{ needs.check-if-main.outputs.is-main != 'true' || contains(github.ref_name, '-') }}
-          make_latest: ${{ !(github.base_ref != 'main' || contains(github.ref_name, '-')) && steps.is_latest.outputs.value }}
+            Release ${{ github.ref_name }} from workflow [#${{ github.run_id }}](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})
+          draft: ${{ needs.check-main.outputs.is-main != 'true' || contains(github.ref_name, '-') }}
+          make_latest: ${{ !(needs.check-main.outputs.is-main != 'true' || contains(github.ref_name, '-')) && steps.is_latest.outputs.value }}
```

Comment:
Since the condition `needs.check-main.outputs.is-main != 'true' || contains(github.ref_name, '-') ` is repeated, and the expressions are a little bit dense to read, could they be broken out to an `env:` var?

Reply:
Will do!

Reply:
Looking at [context availability](https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability), it seems the env context is not available with job level if.

However I found some way around it by setting the condition as an env variable on publish and using it as an output on the following step that depends on it.